website/docs/user_security.md

```diff
@@ -0,0 +1,294 @@
+
+```python
+class UserSecurity(ABC):
+    def __init__(self, cookie_name: str, token_handler: TokenHandler):
+```
+`cookie_name` (*str*) : name of the cookie in which the token is passed
+
+`token_handler` : expects a `TokenHandler` class that implements the `encode` and `decode` functions.
+
+>`UserSecurity` token handler uses the `JWTTokenHandler` in `Core-Lib` which handles the `jwt` tokens.
+
+
+
+###Function provides by UserSecurity
+
+- `secure_entry` is an abstract method that the user must implement, it can be customized to perform actions according to the `policies` supplied to the function.
+
+```python
+def secure_entry(self, request, session_obj, policies: list):
+```
+`request` :  the received request object.
+
+`session_obj` : decoded and formatted session data.
+
+`policies` (*list*) : list of policies that will be required for further authentication or authorization.
+
+- `from_session_data` also an abstract method to be implemented by the user, takes care of the formatting and 
+cleaning of the decoded session data.
+
+```python
+def from_session_data(self, session_data: dict):
+```
+`session_data` (*dict*) : decoded session data received from `_secure_entry()`.
+
+- `generate_session_data` also an abstract method to be implemented by user, returns a structured `dict` with the received data that will be used in the response object.
+
+```python
+def generate_session_data(self, obj) -> dict:
+```
+`obj` : data that must be structured and returned.
+
+- `generate_session_data_token` is a method that will encode the data returned by `generate_session_data()` and return the encoded token.
+
+```python
+def generate_session_data_token(self, obj):
+```
+`obj` that is being passed to `generate_session_data` in order to create a structured `dict`.
+
+- `_secure_entry` is being called in the `@RequireLogin` decorator and is responsible for calling the `secure_entry` 
+method that is implemented.
+
+```python
+def _secure_entry(self, request, policies):
+```
+`request` : request object that is received by the decorator containing the cookie with token.
+
+`policies` : list of policies that will be passed to `secure_entry()`
+
+
+## SecurityHandler
+`SecurityHandler` class registers our `UserSecurity` implemented class and is used to call `UserSecurity` methods using `get()`.
+
+```python
+class SecurityHandler(object):
+```
+
+###Function provides by SecurityHandler
+
+- `register` this function registers our `UserSecurity` implemented class.
+
+```python
+def register(user_security: UserSecurity):
+```
+`user_security` : `UserSecurity` implemented class.
+
+- `get` this function returns the `UserSecurity` functions.
+
+```python
+def get() -> UserSecurity:
+```
+
+
+## RequireLogin Decorator
+This decorator with be responsible for authorization or authentication using `UserSecurity` functions and `SecurityHandler`.

```

Comment:
Grammarly Again

Reply:
Grammarly is not showing any errors

Reply:
<img width="1265" alt="image" src="https://user-images.githubusercontent.com/5166849/157797584-1217ef57-c7b0-4fc6-9eb6-35d978969e88.png">